pkg/blockstorage/zone/zone.go

```diff
@@ -222,6 +228,65 @@ func NodeZonesAndRegion(ctx context.Context, cli kubernetes.Interface) (map[stri
 	return zoneSet, region[0], nil
 }
 
+// GetReadySchedulableNodes addresses the common use case of getting nodes you can do work on.
+// 1) Needs to be schedulable.
+// 2) Needs to be ready.
+// Derived from "k8s.io/kubernetes/test/e2e/framework/node"
+// TODO: check for taints as well
+func GetReadySchedulableNodes(cli kubernetes.Interface) (*v1.NodeList, error) {
+	ns, err := cli.CoreV1().Nodes().List(metav1.ListOptions{FieldSelector: fields.Set{
+		"spec.unschedulable": "false",
+	}.AsSelector().String()})
+	if err != nil {
+		return nil, err
+	}
+	Filter(ns, func(node v1.Node) bool {
+		return IsNodeSchedulable(&node)
+	})
+	if len(ns.Items) == 0 {
+		return nil, errors.New("There are currently no ready, schedulable nodes in the cluster")
```

Comment:
It would be useful to know which nodes were filtered out and why if we are hitting this case (maybe just stats - similar to what is logged in K8s events for pods e.g. 3 found, 2 were unschedulable, 1 was not ready)